### PR TITLE
Update config files for picolibc

### PR DESCRIPTION
**Pull request by mplatings**
* Rename config files
  * No suffix to use picolibc's crt0.o
  * semihost suffix to use picolibc's crt0-semihost.o
  * semihost_ldscript suffix to use picolibc's crt0-semihost.o and
    linker script.
* Patch linker scripts provided by picolibc to work with lld
* Fix --target being specified twice in the config files
* Remove old linker scripts
* Change smoke tests to use semihost_ldscript config
* Fix samples & remove workarounds that are no longer required

### CODE REVIEW

File: samples/ldscripts/microbit.ld [hunk 1]
```diff
@@ -1,40 +1,40 @@
...
  *   FLASH.ORIGIN: starting address of flash
  *   FLASH.LENGTH: length of flash
  *   RAM.ORIGIN: starting address of RAM bank 0
  *   RAM.LENGTH: length of RAM bank 0
  */
 MEMORY
 {
-  FLASH (rx) : ORIGIN = 0x00000000, LENGTH = 0x40000
-  RAM (rwx)  : ORIGIN = 0x20000000, LENGTH = 0x4000
+    FLASH (rx) : ORIGIN = 0x00000000, LENGTH = 0x40000
```
Comment (voltur01) [inline samples/ldscripts/microbit.ld:10]: picolibc.ld seems to provide a way to override the addressed and sizes, e.g.

flash (rx) : ORIGIN = DEFINED(__flash) ? __flash : 0x10000000, LENGTH = DEFINED(__flash_size) ? __flash_size : 0x10000

would it be possible to just use picolibc.ld with correct addresses specified elsewhere?

File: samples/src/baremetal-semihosting/Makefile [hunk 2]
```diff
@@ -20,7 +20,7 @@ include ../../Makefile.conf
 build: hello.elf
 
 hello.elf: *.c
-	$(BIN_PATH)/clang --config armv6m_soft_nofp_semihost_baremetal -g -T ../../ldscripts/microbit.ld -o hello.elf ../../startup/startup_ARMCM0.S $^
+	$(BIN_PATH)/clang --config armv6m_soft_nofp_semihost -g -T ../../ldscripts/microbit.ld -o hello.elf ../../startup/startup_ARMCM0.S $^
```
Comment (voltur01) [inline samples/src/baremetal-semihosting/Makefile:24]: Do we still need the "-m 1073741824k" workaround in the run command with picolibc?

Comment (mplatings) [reply samples/src/baremetal-semihosting/Makefile:24]: We do not :)

**Review by voltur01 — APPROVED** [review]
LGTM, thank you!

**Pull request merged**